staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -210,14 +210,15 @@ func RESTClientFor(config *Config) (*RESTClient, error) {
 	}
 
 	var httpClient *http.Client
+	var clientTimeout time.Duration
 	if transport != http.DefaultTransport {
 		httpClient = &http.Client{Transport: transport}
 		if config.Timeout > 0 {
-			httpClient.Timeout = config.Timeout
+			clientTimeout = config.Timeout
```

Comment:
I think the clientTimeout should be passed to NewRESTClient unconditionally, not only when `transport != http.DefaultTransport`.